lib-find-meal/src/main/java/com/kinandcarta/lib/find/meal/view/FindMealActivity.kt

```diff
@@ -17,52 +27,148 @@ import kotlinx.coroutines.launch
 
 class FindMealActivity : AppCompatActivity() {
     private val mainScope = MainScope()
+    private val sdk = MealsSDK()
+    private val mealsAdapter = MealsAdapter(listOf())
 
     private lateinit var mealsRecyclerView: RecyclerView
     private lateinit var progressBarView: FrameLayout
     private lateinit var swipeRefreshLayout: SwipeRefreshLayout
 
-    private val sdk = MealsSDK()
-    private val mealsAdapter = MealsAdapter(listOf())
+    private lateinit var fusedLocationProviderClient: FusedLocationProviderClient
+    private lateinit var locationCallback: LocationCallback
+    private lateinit var locationRequest: LocationRequest
+    private var locationUpdateState = false
+    private lateinit var lastLocation: Location
```

Comment:
Hmm this feels risky since we can't easily guarantee that lastLocation will be available before it's first used (basically the same as force-unwrapping a property in Swift). I'd probably just make this `var lastLocation: Location? = null`, however wondering if you can just get the last location directly from the client anyway?